clients/admin-ui/src/features/common/tablev2/FidesTable.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@fidesui/react";
+import {
+  flexRender,
+  RowData,
+  Table as TableInstance,
+} from "@tanstack/react-table";
+import React, { ReactNode } from "react";
+
+const getTableTHandTDStyles = (cellId: string) =>
+  cellId === "select"
+    ? { padding: "0px" }
+    : {
+        paddingLeft: "16px",
+        paddingRight: "8px",
+        paddingTop: "0px",
+        paddingBottom: "0px",
+      };
+
+/*
+  This was throwing a false positive for unused paramaters.
+  It's also how the library author reccomends typing meta.
+  https://tanstack.com/table/v8/docs/api/core/column-def#meta
+*/
+/* eslint-disable */
+declare module "@tanstack/table-core" {
+  interface ColumnMeta<TData extends RowData, TValue> {
+    width?: string;
+  }
+}
+/* eslint-enable */
+
+type Props<T> = {
+  tableInstance: TableInstance<T>;
+  rowActionBar?: ReactNode;
+  footer?: ReactNode;
+  onRowClick?: (row: T) => void;
+};
+
+export const FidesTableV2 = <T,>({
+  tableInstance,
+  rowActionBar,
+  footer,
+  onRowClick,
+}: Props<T>) => (
+  <TableContainer
+    overflowY="auto"
+    borderBottomWidth="1px"
+    borderBottomColor="gray.200"
+  >
+    <Table
+      variant="unstyled"
+      style={{
+        borderCollapse: "separate",
+        borderSpacing: 0,
+      }}
+    >
+      <Thead
+        position="sticky"
+        top="0"
+        height="36px"
+        zIndex={10}
+        backgroundColor="gray.50"
+      >
+        {tableInstance.getHeaderGroups().map((headerGroup) => (
+          <Tr key={headerGroup.id}>
+            {headerGroup.headers.map((header) => (
+              <Th
+                key={header.id}
+                width={
+                  header.column.columnDef.meta?.width
+                    ? header.column.columnDef.meta.width
+                    : "unset"
+                }
+                borderTopWidth="1px"
+                borderTopColor="gray.200"
+                borderBottomWidth="1px"
+                borderBottomColor="gray.200"
+                borderRightWidth="1px"
+                borderRightColor="gray.200"
+                _first={{
+                  borderLeftWidth: "1px",
+                  borderLeftColor: "gray.200",
+                }}
+                colSpan={header.colSpan}
+                data-testid={`column-${header.id}`}
+                style={getTableTHandTDStyles(header.column.id)}
+              >
+                {flexRender(
+                  header.column.columnDef.header,
+                  header.getContext()
+                )}
+              </Th>
+            ))}
+          </Tr>
+        ))}
+      </Thead>
+      <Tbody>
+        {rowActionBar}
```

Comment:
The row action bar is rendered separatly from the react table state because, while the library provides a pinning API, it only works nicely for rows that match the column schema defined for the table. 

Since this is a visual row that's separate from the table state it was easier to render it this way.